wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -499,6 +494,86 @@ abstract class Wallet
     } yield tx
   }
 
+  override def bumpFee(
+      txId: DoubleSha256DigestBE,
+      newFeeRate: FeeUnit): Future[Transaction] = {
+    for {
+      txDbOpt <- transactionDAO.findByTxId(txId)
+      tx <- txDbOpt match {
+        case Some(db) => Future.successful(db.transaction)
+        case None =>
+          Future.failed(
+            new RuntimeException(s"Unable to find transaction ${txId.hex}"))
+      }
+
+      outPoints = tx.inputs.map(_.previousOutput).toVector
+      spks = tx.outputs.map(_.scriptPubKey).toVector
+
+      utxos <- spendingInfoDAO.findByOutPoints(outPoints)
+      _ = require(utxos.size == tx.inputs.size,
+                  "Can only bump fee for our own transaction")
+      spendingInfos <- FutureUtil.sequentially(utxos) { utxo =>
+        transactionDAO
+          .findByOutPoint(utxo.outPoint)
+          .map(txDbOpt =>
+            utxo.toUTXOInfo(keyManager = keyManager, txDbOpt.get.transaction))
+      }
+
+      _ = {
+        val inputAmount = utxos.foldLeft(CurrencyUnits.zero)(_ + _.output.value)
+
+        val oldFeeRate = newFeeRate match {
+          case _: SatoshisPerByte =>
+            SatoshisPerByte.calc(inputAmount, tx)
+          case _: SatoshisPerKiloByte =>
+            SatoshisPerKiloByte.calc(inputAmount, tx)
+          case _: SatoshisPerVirtualByte =>
+            SatoshisPerVirtualByte.calc(inputAmount, tx)
+          case _: SatoshisPerKW =>
+            SatoshisPerKW.calc(inputAmount, tx)
+        }
+
+        require(oldFeeRate.currencyUnit < newFeeRate.currencyUnit,
+                "Cannot bump to a lower fee")
+      }
+
+      myAddrs <- addressDAO.findByScriptPubKeys(spks)
+      _ = require(myAddrs.nonEmpty, "Must have an output we own")
+
+      changeSpks = myAddrs.flatMap { db =>
+        if (db.path.chain.chainType == HDChainType.Change) {
+          Some(db.scriptPubKey)
+        } else None
+      }
+
+      changeSpk =
+        if (changeSpks.nonEmpty) {
+          // Pick a random change spk
+          Random.shuffle(changeSpks).head
+        } else {
+          // If none are explicit change, pick a random one we own
+          Random.shuffle(myAddrs.map(_.scriptPubKey)).head
+        }
+
+      // Mark old outputs as replaced
+      oldUtxos <- spendingInfoDAO.findDbsForTx(txId)
+      _ <- spendingInfoDAO.updateAll(
+        oldUtxos.map(_.copyWithState(TxoState.DoesNotExist)))
```

Comment:
Why is this going to `TxoState.DoesNotExist` ? Shouldn't this be `PendingConfirmationsReceived`?

Reply:
If it is `PendingConfirmationsReceived` then this transaction will double count to our balance.